test/commands/force/org/scratchOrgCreate.test.ts

```diff
@@ -0,0 +1,314 @@
+    });
+
+    it('will prompt the user for a secret if clientId is provided', async () => {
+      const connectedAppConsumerKey = 'abcdef';
+      const definitionfile = 'myScratchDef.json';
+      const command = await createCommand([
+        '--type',
+        'scratch',
+        '-i',
+        connectedAppConsumerKey,
+        '--definitionfile',
+        definitionfile,
+        '-u',
+        'testProdOrg',
+      ]);
+      scratchOrgCreateStub.restore();
+      stubMethod(sandbox, Org, 'create').resolves(Org.prototype);
+      const prodOrg = stubMethod(sandbox, Org.prototype, 'scratchOrgCreate').resolves({
+        username: 'sfdx-cli@salesforce.com',
+        scratchOrgInfo: {},
+        authFields: {
+          accessToken: '1234',
+          clientId: '1234',
+          created: '2022-01-01',
+          createdOrgInstance: 'instance',
+          devHubUsername: 'sfdx-cli@salesforce.com',
+          expirationDate: '2021-01-01',
+          instanceUrl: 'https://instance.salesforce.com',
+          loginUrl: 'https://login.salesforce.com',
+          orgId: '12345',
+          username: 'sfdx-cli@salesforce.com',
+        },
+        warnings: [],
+      });
+      await command.runIt();
+      expect(prodOrg.firstCall.args[0]).to.deep.equal({
+        apiversion: undefined,
+        clientSecret,
+        connectedAppConsumerKey,
+        definitionfile,
+        definitionjson: undefined,
+        durationDays: undefined,
+        noancestors: undefined,
+        nonamespace: undefined,
+        wait: {
+          quantity: 6,
+          unit: 0,
+        },
+        retry: 0,
+        orgConfig: {},
+      });
+      expect(promptStub.callCount).to.equal(1);
+      expect(uxLogStub.firstCall.firstArg).to.equal(
+        'Successfully created scratch org: 12345, username: sfdx-cli@salesforce.com'
+      );
+    });
+
+    it('will set alias/defaultusername', async () => {
+      const definitionfile = 'myScratchDef.json';
+      const command = await createCommand([
+        '--type',
+        'scratch',
+        '--setalias',
+        'sandboxAlias',
+        '--setdefaultusername',
+        '--definitionfile',
+        definitionfile,
+        '-u',
+        'testProdOrg',
+      ]);
+
+      scratchOrgCreateStub.restore();
+      stubMethod(sandbox, Org, 'create').resolves(Org.prototype);
+      const prodOrg = stubMethod(sandbox, Org.prototype, 'scratchOrgCreate').resolves({
+        username: 'newScratchUsername',
+        scratchOrgInfo: {},
+        authFields: {
+          accessToken: '1234',
+          clientId: '1234',
+          created: '2022-01-01',
+          createdOrgInstance: 'instance',
+          devHubUsername: 'sfdx-cli@salesforce.com',
+          expirationDate: '2021-01-01',
+          instanceUrl: 'https://instance.salesforce.com',
+          loginUrl: 'https://login.salesforce.com',
+          orgId: '12345',
+          username: 'sfdx-cli@salesforce.com',
+        },
+        warnings: [],
+      });
```

Comment:
Nit: I'd consider having a base object for the `resolves` so the tests are a bit easier to read and maintain.